src/controller.rs

```diff
@@ -150,13 +198,52 @@ async fn launch_job(
             backoff_limit: Some(1),
             template: PodTemplateSpec {
                 spec: Some(PodSpec {
+                    volumes: Some(vec![
+                        Volume {
+                            name: "credentials".to_string(),
+                            secret: Some(SecretVolumeSource {
+                                secret_name: Some("gar-docker-secret".to_string()),
```

Comment:
One thing I'm not too sure of is how we can make this generic, i.e. I know we can use the `imagePullSecrets` from the `AppInstance` itself, but there is no guarantee that they are shared between access to a registry like GAR and where the operator containers reside too.

Maybe I'm over-complicating this though and we can discuss further once you're back @mkmik, as I might be a little hazy on some details.